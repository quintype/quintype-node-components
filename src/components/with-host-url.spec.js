import { WithHostUrl } from '..';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render, cleanup } from 'react-testing-library';
import React from 'react';

afterEach(cleanup);

describe("WithHostUrl", () => {
  it("Passes the hosts to the render props", () => {
    const { container } = render(
      <Provider store={createStore(x => x, { qt: { primaryHostUrl: "https://www.foo.com", currentHostUrl: "https://subdomain.foo.com"} })}>
        <WithHostUrl>{({ primaryHostUrl, currentHostUrl }) =>
          <div>
            <div>primaryHostUrl: {primaryHostUrl}</div>
            <div>currentHostUrl: {currentHostUrl}</div>
          </div>
        }</WithHostUrl>
      </Provider>
    )
    expect(container.firstChild).toMatchSnapshot();
  });
})
