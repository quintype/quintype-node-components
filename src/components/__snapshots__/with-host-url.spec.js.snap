// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithHostUrl Passes the hosts to the render props 1`] = `
<div>
  <div>
    primaryHostUrl: 
    https://www.foo.com
  </div>
  <div>
    currentHostUrl: 
    https://subdomain.foo.com
  </div>
</div>
`;
